xbmc/network/GUIDialogNetworkSetup.cpp

```diff
@@ -131,7 +142,12 @@ bool CGUIDialogNetworkSetup::ShowAndGetNetworkAddress(std::string &path)
   CGUIDialogNetworkSetup *dialog = g_windowManager.GetWindow<CGUIDialogNetworkSetup>(WINDOW_DIALOG_NETWORK_SETUP);
   if (!dialog) return false;
   dialog->Initialize();
-  dialog->SetPath(path);
+  if (!dialog->SetPath(path))
```

Comment:
meh, there is a call to `ShowAndGetNetworkAddress` in `CGUIDialogFileBrowser::OnAddNetworkLocation` where an empty path is given to this method. This should open a blank dialog to add a new network location https://github.com/xbmc/xbmc/blob/master/xbmc/dialogs/GUIDialogFileBrowser.cpp#L897

We should return true in `SetPath` if the given path is empty (early return).

Reply:
i'm sure the call for empty was to wipe out previous data, so i did it a bit differently.